File: 0551. Student Attendance Record I/Solution.java
class Solution {
    public boolean checkRecord_MK1(String s) {
        int absent = 0, contin_late = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == 'L')
                contin_late++;
            else {
                contin_late = 0;
                if (s.charAt(i) == 'A')
                    absent++;
            }
            if (absent > 1 || contin_late > 2)
                return false;
        }
        return true;
    }

    public boolean checkRecord_MK2(String s) {
        return s.indexOf('A') == s.lastIndexOf('A') && s.indexOf("LLL") == -1;
    }

    public boolean checkRecord_MK3(String s) {
        return !s.matches(".*(A.*A|LLL).*");
    }
}

File: 0551. Student Attendance Record I/Solution.py
import re


class Solution:
    def checkRecord_MK1(self, s: str) -> bool:
        absent, contin_late = 0, 0
        for ch in s:
            if ch == 'L':
                contin_late += 1
            else:
                contin_late = 0
                if ch == 'A':
                    absent += 1
            if absent > 1 or contin_late > 2:
                return False
        return True

    def checkRecord_MK2(self, s: str) -> bool:
        return s.count('A') <= 1 and s.count('LLL') == 0

    def checkRecord_MK3(self, s: str) -> bool:
        return not re.search('A.*A|LLL', s)
